packages/components/src/box-control/input-controls.js

```diff
@@ -91,25 +92,35 @@ export default function BoxInputControls( {
 				align="top"
 				className="component-box-control__input-controls"
 			>
-				{ filteredSides.map( ( side ) => (
-					<UnitControl
-						{ ...props }
-						isFirst={ first === side }
-						isLast={ last === side }
-						isOnly={ only === side }
-						value={ values[ side ] }
-						unit={
```

Comment:
For ease of review: the `unit` prop here was removed, and instead the parsing and potentially overriding of the unit is applied to the `value` prop